proof/refine/ARM/Schedule_R.thy

```diff
@@ -3864,6 +3865,114 @@ lemma awaken_corres:
+     apply (case_tac z; clarsimp simp: obj_at'_def projectKOs)
+    apply (erule cross_relF [OF _ sc_at'_cross_rel], clarsimp simp: obj_at_def is_sc_obj)
+    apply (erule (1) valid_sched_context_size_objsI)
+   apply (case_tac z; clarsimp simp: obj_at'_def projectKOs)
+  apply (subgoal_tac "tcb_at t s")
+   apply (frule cross_relF [OF _ tcb_at'_cross_rel], clarsimp, assumption)
+   apply (clarsimp simp: obj_at'_def projectKOs)
+  apply (clarsimp simp: obj_at_def vs_all_heap_simps pred_map_def is_tcb)
+  done
+
+lemma setNextInterrupt_corres:
+  "corres dc ((\<lambda>s. active_sc_tcb_at (cur_thread s) s) and valid_release_q and valid_objs
+              and active_sc_valid_refills and pspace_aligned and pspace_distinct)
+             valid_objs'
+             set_next_interrupt
+             setNextInterrupt"
+  unfolding setNextInterrupt_def set_next_interrupt_def
+  apply (rule stronger_corres_guard_imp)
+    apply (rule corres_split [OF getCurTime_corres])
+      apply (rule corres_split [OF gct_corres], simp)
+        apply (rule corres_split_eqr [OF _ get_tcb_obj_ref_corres])
+           apply (rule corres_assert_opt_assume_l)
+           apply (rule corres_split [OF get_sc_corres])
+             apply (rule_tac F="standard_sc rv'" in corres_gen_asm2)
+             apply (rule corres_split [OF corres_if])
+                  apply (clarsimp simp: num_domains_def numDomains_def)
+                 apply (rule corres_split [OF domain_time_corres])
+                   apply (simp only: fun_app_def)
+                   apply (rule corres_return_eq_same)
+                   apply (clarsimp simp: refills_heads_equal refill_map_def)
+                  apply wpsimp
+                 apply wpsimp
+                apply (rule corres_return_eq_same)
+                apply (clarsimp simp: refills_heads_equal refill_map_def)
+               apply (rule corres_split [OF getReleaseQueue_corres])
+                 apply (rule corres_split [OF corres_if], simp)
+                     apply (rule corres_return_eq_same, simp)
+                    apply (rule corres_split_eqr)
+                       apply (rule corres_assert_opt_assume_l)
+                       apply (simp)
+                       apply (rule corres_split [OF get_sc_corres])
+                         apply (rule_tac F="standard_sc rv'c" in corres_gen_asm2)
+                         apply (rule corres_return_eq_same)
+                         apply (clarsimp simp: refills_heads_equal refill_map_def)
+                        apply wpsimp
+                       apply wpsimp
+                      apply (simp, rule get_tcb_obj_ref_corres)
+                      apply (clarsimp simp: tcb_relation_def)
+                     apply (wpsimp wp: get_tcb_obj_ref_wp)
+                    apply (wpsimp wp: threadGet_wp)
+                   apply (rule corres_machine_op)
+                   apply (simp only:, rule setDeadline_corres, simp)
```

Comment:
Out of curiosity, what is that `simp only:` needed for?

Reply:
Good point, I've changed it to use `simp del: dc_simp`.